lib/msal-common/src/authority/Authority.ts

```diff
@@ -438,33 +443,46 @@ export class Authority {
      * @param newMetadata
      */
     private async updateCloudDiscoveryMetadata(metadataEntity: AuthorityMetadataEntity): Promise<AuthorityMetadataSource> {
+        this.logger.verbose("Attempting to get cloud discovery metadata in the config");
         let metadata = this.getCloudDiscoveryMetadataFromConfig();
         if (metadata) {
+            this.logger.verbose(`Found the following cloud discovery metata in the config: ${JSON.stringify(metadata)}`);
             metadataEntity.updateCloudDiscoveryMetadata(metadata, false);
             return AuthorityMetadataSource.CONFIG;
         }
+        this.logger.verbose("Did not find cloud discovery metadata in the config");
 
-        // If The cached metadata came from config but that config was not passed to this instance, we must go to the network
+        // If the cached metadata came from config but that config was not passed to this instance, we must go to the network
+        this.logger.verbose("Attempting to get cloud discovery metadata in the cache");
         if (this.isAuthoritySameType(metadataEntity) && metadataEntity.aliasesFromNetwork && !metadataEntity.isExpired()) {
+            this.logger.verbose(`Found the following metadata in the cache: ${JSON.stringify(metadata)}`);
             // No need to update
             return AuthorityMetadataSource.CACHE;
         }
+        this.logger.verbose("Did not find cloud discovery metadata in the cache");
 
-        const harcodedMetadata = this.getCloudDiscoveryMetadataFromHarcodedValues();
-
+        this.logger.verbose("Attempting to get cloud discovery metadata from the network");
         metadata = await this.getCloudDiscoveryMetadataFromNetwork();
         if (metadata) {
+            this.logger.verbose(`Found the following cloud discovery metata from the network: ${JSON.stringify(metadata)}`);
             metadataEntity.updateCloudDiscoveryMetadata(metadata, true);
             return AuthorityMetadataSource.NETWORK;
         }
+        this.logger.verbose("Did not find cloud discovery metadata from the network");
         
+        this.logger.verbose("Attempting to get cloud discovery metadata from hardcoded values");
+        const harcodedMetadata = this.getCloudDiscoveryMetadataFromHarcodedValues();
         if (harcodedMetadata && !this.options.skipAuthorityMetadataCache) {
+            this.logger.verbose(`Found the following cloud discovery metata from hardcoded values: ${JSON.stringify(metadata)}`);
```

Comment:
Same observation as above - if verbose ...

Reply:
the logger already functions like this. There are different tiers of logging: info, trace, verbose, etc. The logger can be found in /lib/msal-common/src/logger/Logger.ts. By using .verbose here, this message will only print when the user has specified "verbose" in the logging configuration. 

For example:
```
const clientConfig: msal.Configuration = {
    system: {
        loggerOptions: {
            loggerCallback(loglevel, message, containsPii) {
                console.log(message);
            },
            piiLoggingEnabled: false,
            logLevel: msal.LogLevel.Verbose,
        },
```

Reply:
I think the point here is that `stringify` will always get called first to create the message string to pass into the `verbose` regardless whether verbose logging is enabled or not (since that's checked later).